nacos-config-spring-boot-autoconfigure/src/main/java/com/alibaba/boot/nacos/config/util/log/LogAutoFreshProcess.java

```diff
@@ -0,0 +1,139 @@
+
+    private final NacosConfigProperties nacosConfigProperties;
+
+    private final ConfigurableEnvironment environment;
+
+    private static final List<String> LOG_DATA_ID = new ArrayList<>();
+
+    private static final String LOG_CACHE_BASE = System.getProperty("JM.SNAPSHOT.PATH", System.getProperty("user.home")) + File.separator + "nacos"
+            + File.separator + "logConfig";
+
+    static {
+        LOG_DATA_ID.add("logback.xml");
+        LOG_DATA_ID.add("log4j2.xml");
+    }
+
+    public LogAutoFreshProcess(ConfigurableEnvironment environment, NacosConfigProperties nacosConfigProperties) {
+        this.nacosConfigProperties = nacosConfigProperties;
+        this.environment = environment;
+    }
+
+    public void process(Function<Properties, ConfigService> builder) {
+        final String groupName = environment
+                .resolvePlaceholders(nacosConfigProperties.getGroup());
+        for (String dataId : LOG_DATA_ID) {
+            ConfigService configService = builder.apply(buildGlobalNacosProperties());
+            String content = NacosUtils.getContent(configService, dataId, groupName);
+            if (StringUtils.isNotBlank(content)) {
+                writeLogFile(content, dataId);
+                System.setProperty(LoggingApplicationListener.CONFIG_PROPERTY, LOG_CACHE_BASE + File.separator + dataId);
+                registerListener(configService, dataId, groupName);
+            }
+        }
+    }
+
+    public Properties buildGlobalNacosProperties() {
+        return NacosPropertiesBuilder.buildNacosProperties(environment,
+                nacosConfigProperties.getServerAddr(),
+                nacosConfigProperties.getNamespace(), nacosConfigProperties.getEndpoint(),
+                nacosConfigProperties.getSecretKey(),
+                nacosConfigProperties.getAccessKey(),
+                nacosConfigProperties.getRamRoleName(),
+                nacosConfigProperties.getConfigLongPollTimeout(),
+                nacosConfigProperties.getConfigRetryTime(),
+                nacosConfigProperties.getMaxRetry(),
+                nacosConfigProperties.isEnableRemoteSyncConfig(),
+                nacosConfigProperties.getUsername(), nacosConfigProperties.getPassword());
+    }
+
+    private void registerListener(ConfigService configService, String dataId, String groupName) {
+        try {
+            configService.addListener(dataId, groupName, new AbstractListener() {
+                @Override
+                public void receiveConfigInfo(String configInfo) {
+                    if (StringUtils.isNotBlank(configInfo)) {
+                        writeLogFile(configInfo, dataId);
+                    }
+                }
+            });
+        } catch (NacosException e) {
+            throw new RuntimeException(
+                    "ConfigService can't add Listener with dataId : " + dataId,
+                    e);
+        }
+
+    }
+
+    private void writeLogFile(String content, String dataId) {
```

Comment:
Nacos client will cache config in cache dir, why cache agqin?

Reply:
> Nacos client will cache config in cache dir, why cache agqin?
The purpose of writing the configuration content into the file here is to facilitate the log framework to read from the directory, because the snapshot file path of the Nacos client is not exposed. If the dead path written on the springboot side points to the Nacos client, problems will occur later when the Nacos client is changed.